### PR TITLE
Change schema table from horizontal to vertical

### PR DESCRIPTION
**Pull request by cafenoctua**
クエリの結果ウィンドウで表示されるスキーマを横から縦に変更してみました。

### CODE REVIEW

**Comment by minodisk** [comment]
First of all, thank you for sending me the PR 🚀 
The code itself in this PR is good.

I would appreciate it if you could explain in the PR why horizontal table is not good or why vertical table is necessary, which cannot be expressed in the code.
Because after merging this PR, if a PR comes in that says, for example, "Changed the schema table from vertical to horizontal", I will not be able to decide whether I should merge that PR or not.

---

For example, but one reason I can think of is something like, "If we add a description column to the schema in the future, the vertical table should be easier to read".

image

**Comment by cafenoctua** [comment]
Thank you for your explanation.
The purpose of this addition was to improve visibility even when there were many rows.
Also, as you say, I implemented it because I thought it would be easier to see if I added a description column.

File: packages/viewer/src/domain/Schema.tsx [hunk 1]
```diff
@@ -15,27 +15,21 @@ export const Schema: FC<{
...
+          <Tr>
+            <Tr>
+              <Th>Column</Th>
+              <Th>Type</Th>
+              <Th>Mode</Th>
+            </Tr>
+            {heads.map(({ id, type, mode }) => (
+              <Tr><Th key={id}>{id}</Th>
+              <Th key={id}>{type}</Th>
+              <Th key={id}>{mode}</Th></Tr>
```
Comment (minodisk) [inline packages/viewer/src/domain/Schema.tsx:29]: I think the `key` works when set to the parent element.

```jsx
<Tr key={id}>
  <Th>{id}</Th>
  <Th>{type}</Th>
  <Th>{mode}</Th>
</Tr>
```

File: packages/viewer/src/domain/Schema.tsx [hunk 2]
```diff
@@ -15,27 +15,21 @@ export const Schema: FC<{
...
+            ))}
+          </Tr>
+        </Thead>
```
Comment (minodisk) [inline packages/viewer/src/domain/Schema.tsx:32]: A `THead` is set for a child of `TBody`; the `THead` and `TBody` must be placed in the same hierarchy as children of `Table`:

```jsx
<Tbody>
  <Thead>
  </Thead>
  <TBody>
  </Tbody>
</Table>
```

See [\<table\>: The Table element \- HTML: HyperText Markup Language \| MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table).

**Review by minodisk — CHANGES_REQUESTED** [review]

**Comment by cafenoctua** [comment]
Thank you for your review
I understood what you meant, so I fixed it.

File: packages/viewer/src/domain/Schema.tsx [hunk 3]
```diff
@@ -15,21 +15,23 @@ export const Schema: FC<{
...
-            {heads.map(({ id, type, mode }) => (
-              <Tr><Th key={id}>{id}</Th>
-              <Th key={id}>{type}</Th>
-              <Th key={id}>{mode}</Th></Tr>
-            ))}
+            <Th>Column</Th>
+            <Th>Type</Th>
+            <Th>Mode</Th>
+          </Tr>
+        </Tr>
```
Comment (minodisk) [inline packages/viewer/src/domain/Schema.tsx:25]: `Tr`s are duplicated. Must be only one.

**Review by minodisk — CHANGES_REQUESTED** [review]
One more 🏃‍♂️

File: packages/viewer/src/domain/Schema.tsx (hunk 3 above)
Comment (cafenoctua) [reply packages/viewer/src/domain/Schema.tsx:23]: Oops
Deleted unnecessary Tr tags.

**Review by minodisk — APPROVED** [review]
LGTM 👍

**Comment by minodisk** [comment]
Released as [v1.12.0](https://github.com/minodisk/bigquery-runner/blob/main/CHANGELOG.md#changed) 🎉 

FYI, there was another feature released at the same time that allowed multiple tables handled in child jobs to be displayed as separate table tabs, and the schema tab was merged into the table tab because of that.

image

**Pull request merged**